libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1417,16 +1417,21 @@ void WaveTrack::ClearAndPasteOne(
 
    // Restore cut/split lines
    if (preserve) {
-      auto attachLeft = [](WaveClip &target, WaveClip &src)
-      {
-         assert(target.GetWidth() == src.GetWidth());
+      auto attachLeft = [](WaveClip& target, WaveClip& src) {
+         // What this lambda does is restoring the left hidden data of `target`
+         // that was cleared by `HandleClear`. Hence, `target` has no left
+         // hidden data at this stage.
          assert(target.GetTrimLeft() == 0);
          if (target.GetTrimLeft() != 0)
             return;
 
+         // `src` was created by copy from `target`, so they have equal width
+         // and stretch ratio.
+         assert(target.GetWidth() == src.GetWidth());
+         assert(target.GetStretchRatio() == src.GetStretchRatio());
```

Comment:
Not just commenting assertions but adding one.  So can you prove this new one is satisfied at the point of the call?

Reply:
Provided what the comment I added is correct, yes.